diff -Naur tcpip_old/Makefile tcpip/Makefile
--- tcpip_old/Makefile	2005-08-05 16:49:23.367528064 +0000
+++ tcpip/Makefile	2005-08-05 16:58:46.346942184 +0000
@@ -20,7 +20,10 @@
 
 LWIP=../lwip
 
-ps2ip_OBJECTS = ps2ip.o sys.o sys_arch.o exports.o imports.o inet.o mem.o netif.o pbuf.o stats.o tcp_in.o tcp_out.o udp.o memp.o tcp.o etharp.o raw.o
+ps2ip_OBJECTS = ps2ip.o sys.o sys_arch.o exports.o imports.o inet.o mem.o \
+	netif.o pbuf.o stats.o tcp_in.o tcp_out.o udp.o memp.o tcp.o \
+	etharp.o raw.o \
+	hdld_svr.o hio_iop.o byteseq.o
 
 ifdef PS2IP_DHCP
 ps2ip_OBJECTS+= dhcp.o
@@ -46,6 +49,7 @@
 IOP_CFLAGS += -Wall -fno-builtin-printf -DLWIP_NOASSERT $(DEBUG_FLAGS)
 IOP_LDFLAGS += -s
 
+IOP_CFLAGS += -D_BUILD_PS2 -I../../../
 
 all: $(IOP_OBJS_DIR) $(IOP_BIN_DIR) $(IOP_BIN)
 
@@ -123,6 +127,12 @@
 obj/etharp.o: $(LWIP)/src/netif/etharp.c
 	$(IOP_CC) $(IOP_CFLAGS) $< -o $@
 
+obj/hio_iop.o: ../../iop/hio_iop.c
+	$(IOP_CC) $(IOP_CFLAGS) $< -o $@
+
+obj/byteseq.o: ../../../byteseq.c
+	$(IOP_CC) $(IOP_CFLAGS) $< -o $@
+
 ifdef PS2IP_DHCP
 obj/dhcp.o: $(LWIP)/src/core/dhcp.c
 	$(IOP_CC) $(IOP_CFLAGS) $< -o $@
diff -Naur tcpip_old/src/imports.lst tcpip/src/imports.lst
--- tcpip_old/src/imports.lst	2005-08-05 16:47:21.000000000 +0000
+++ tcpip/src/imports.lst	2005-07-31 16:49:19.000000000 +0000
@@ -36,7 +36,9 @@
 sysclib_IMPORTS_start
 I_memset
 I_strcpy
+I_strlen
 I_memcpy
+I_memcmp
 sysclib_IMPORTS_end
 
 sysmem_IMPORTS_start
@@ -46,4 +48,16 @@
 
 stdio_IMPORTS_start
 I_printf
+I_sprintf
 stdio_IMPORTS_end
+
+atad_IMPORTS_start
+I_ata_get_devinfo
+I_ata_device_dma_transfer
+I_ata_device_flush_cache
+atad_IMPORTS_end
+
+dev9_IMPORTS_start
+I_dev9Shutdown
+I_dev9IntrDisable
+dev9_IMPORTS_end
diff -Naur tcpip_old/src/irx_imports.h tcpip/src/irx_imports.h
--- tcpip_old/src/irx_imports.h	2005-08-05 16:47:21.000000000 +0000
+++ tcpip/src/irx_imports.h	2005-07-31 14:59:43.000000000 +0000
@@ -17,6 +17,8 @@
 #include "irx.h"
 
 /* Please keep these in alphabetical order!  */
+#include "atad.h"
+#include "dev9.h"
 #include "intrman.h"
 #include "loadcore.h"
 #include "stdio.h"
diff -Naur tcpip_old/src/ps2ip.c tcpip/src/ps2ip.c
--- tcpip_old/src/ps2ip.c	2005-08-05 16:49:23.429518640 +0000
+++ tcpip/src/ps2ip.c	2005-08-05 16:55:08.881002032 +0000
@@ -414,6 +414,8 @@
 }
 
 
+void hdld_svr_init (void);
+
 int
 _start(int argc,char** argv)
 {
@@ -452,5 +454,7 @@
 
 	dbgprintf("PS2IP: System Initialised\n");
 
+	hdld_svr_init ();
+
 	return	iRet; 
 }
